src/webnn_native/xnnpack/GraphXNN.cpp

```diff
@@ -743,30 +750,43 @@ namespace webnn_native::xnnpack {
     }
 
     MaybeError Graph::ComputeImpl(NamedInputsBase* inputs, NamedOutputsBase* outputs) {
-        std::vector<xnn_external_value> externalValues;
-        for (auto& input : inputs->GetRecords()) {
-            if (mExternals.find(input.first) == mExternals.end()) {
-                return DAWN_VALIDATION_ERROR("Invalid input.");
+        if (mNamedInputs != inputs || mNamedOutputs != outputs) {
+            bool anyPointersChanged = false;
+            for (auto& input : inputs->GetRecords()) {
+                if (mExternals.find(input.first) == mExternals.end()) {
```

Comment:
There is a macro named DAWN_INVALID_IF that can simplify the usage.

Reply:
fixed in https://github.com/webmachinelearning/webnn-native/pull/249/commits/1ae8d6501923154a085218b43f45ffbef3e081c8, please take another look. Thanks.